src/contract/event.rs

```diff
@@ -465,6 +473,14 @@ impl<T: Transport, E: ParseLog> AllEventsBuilder<T, E> {
         self
     }
 
+    /// Limit the number of events that can be retrieved by this filter.
+    ///
+    /// Note that this parameter is non-standard.
+    pub fn limit(mut self, value: usize) -> Self {
+        self.filter = self.filter.limit(value);
+        self
+    }
+
```

Comment:
I feel like this entire block is duplicated...

Reply:
It's unfortunate but we currently need two separate types for building a filter for all events and for a single event, since they have slightly different semantics.

Here is a link to the original discussion: https://github.com/gnosis/ethcontract-rs/pull/250#discussion_r400055786

While I plan on improving on this code in the future, it is a necessary evil for now :cry:. A code smell if you will.